include/pinocchio/math/quaternion.hpp

```diff
@@ -22,6 +19,24 @@ namespace pinocchio
 {
   namespace quaternion
   {
+    template <typename Scalar>
+    struct DefaultNormTolerance
+    {
+        static constexpr Scalar value() { return Scalar(1e-8); } // Generic fallback
```

Comment:
We should not provide default value a priori for raising proper compilation issues if not specialized for a given type

Reply:
At the moment, when we have a `CppAD::AD<BaseScalar>` we fall back to the generic tolerance. This is for sure assuming that `BaseScalar` is a double-based type (e.g., `CppAD::cg::CG<double>`).

I agree that this is not the full solution, but I wanted to get this first PR done. Part of this is to unlock Crocoddyl development, another part is to test later with `CppAD::cg::CG<float>`.

I think a similar story would be for Casadi. But I can't tell more. This is because I'm not familiar with Casadi as you are.

Reply:
Please let me know what you suggest. Not yet super clear what I should do here. Thanks!

Reply:
I removed the default tolerance and added a specialization for `CppAD::cg::GG` types. This approach can handle CppADCG types for both doubles and floats. I'll mark this conversation as resolved!